examples/api/engineering_formatter.py

```diff
@@ -14,13 +14,30 @@
 # Fixing random state for reproducibility
 prng = np.random.RandomState(19680801)
 
-fig, ax = plt.subplots()
-ax.set_xscale('log')
-formatter = EngFormatter(unit='Hz')
-ax.xaxis.set_major_formatter(formatter)
-
+# Create artificial data to plot.
+# The x data span over several decades to demonstrate several SI prefixes.
 xs = np.logspace(1, 9, 100)
 ys = (0.8 + 0.4 * prng.uniform(size=100)) * np.log10(xs)**2
-ax.plot(xs, ys)
+
+# Figure width is doubled (2*6.4) to display nicely 2 subplots side by side.
+fig, (ax0, ax1) = plt.subplots(ncols=2, figsize=(12.8, 4.8))
+for ax in (ax0, ax1):
+    ax.set_xscale('log')
+
+# Demo of the default settings, with a user-defined unit label.
+ax0.set_title('Full unit ticklabels, w/ default precision & space separator')
+formatter0 = EngFormatter(unit='Hz')
+ax0.xaxis.set_major_formatter(formatter0)
+ax0.plot(xs, ys)
+ax0.set_xlabel('Frequency')
+
+# Demo of the options `places` (number of digit after decimal point) and
+# `sep` (separator between the number and the prefix/unit).
+ax1.set_title('SI-prefix only ticklabels, 1-digit precision & ' +
+              'hair space separator')
+formatter1 = EngFormatter(places=1, sep=u"\u200a")  # U+200A is hair space
```

Comment:
You should use a thin space (https://en.wikipedia.org/wiki/Thin_space).  "\N{THIN SPACE}" (or "\N{HAIR SPACE}") is also more self-documenting.